loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from datetime import date, timedelta
+
+from odoo.exceptions import UserError
+from odoo.tests import TransactionCase, tagged
+
+from odoo.addons.base.tests.common import DISABLED_MAIL_CONTEXT
+
+
+@tagged("-at_install", "post_install")
+class LoyaltyInitialDateValidity(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, **DISABLED_MAIL_CONTEXT))
+        loyalty_program = cls.env["loyalty.program"]
+        # Promotion with start date currently valid
+        cls.promotion = loyalty_program.create(
+            {
+                "name": "Test loyalty initial date validity",
+                "program_type": "promotion",
+                "trigger": "auto",
+                "applies_on": "current",
+                "date_from": date.today() - timedelta(days=1),
+                "rule_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "reward_point_mode": "order",
+                            "minimum_qty": 1,
+                        },
+                    )
+                ],
+                "reward_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "reward_type": "discount",
+                            "discount": 10,
+                            "discount_mode": "percent",
+                            "discount_applicability": "order",
+                        },
+                    )
+                ],
+            }
+        )
+
+    def test_01_check_date_from_date_to(self):
+        # import wdb; wdb.set_trace()
+        self.assertFalse(self.promotion.date_to)
+        # The end date cannot be earlier than the start date.
+        with self.assertRaises(UserError):
+            self.promotion.date_to = date.today() - timedelta(days=7)
+        self.promotion.date_to = date.today() + timedelta(days=7)
```

Comment:
Relative dates in tests are hard to read. It's better to use `freezegun` and fixed dates.

Reply:
done, thanks!